File: Data Structures & Algorithms/top-k-elements-in-list/submission-17.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let count = {};

        for (const n of nums) {
            count[n] = (count[n] || 0) + 1;
        }

        let arr = Object.entries(count).map(([num, freq]) => [parseInt(num), freq]);
        arr.sort((a, b) => b[1] - a[1]);
        return arr.slice(0, k).map((pair) => pair[0]);
    }
}

File: Data Structures & Algorithms/is-anagram/submission-17.go
func isAnagram(s string, t string) bool {
    if len(s) != len(t) {
        return false
    }

    count := [26]int{}
    for i := range s {
        count[s[i] - 'a']++
        count[t[i] - 'a']--
    }

    for _, num := range count {
        if num != 0 {
            return false
        }
    }
    return true
}
